main.nf

```diff
@@ -463,54 +529,97 @@ def SHOW_USEAGE() {
         if (helpFile.exists()) {
             println helpFile.text
         } else {
-            println "Sorry something went wrong, usage file not found! please vist our website for more info : https://ai-marrvel.readthedocs.io/en/latest/"
+            println """
+            Sorry something went wrong, usage file not found!
+            Please vist our website for more info : https://ai-marrvel.readthedocs.io/en/latest/
+            """
         }
         exit 0
     }
 }
 
+process VALIDATE_INPUT_PARAMS {
+    // Later will publish there validation error file into its own directory for webApp use.
+    input:
+    path vcf
+    path hpo
+    path refDir
+
+    output:
+    val isValidationPassed
+
+    script:
+    def userParamsJson = groovy.json.JsonOutput.toJson(params)
+    isValidationPassed = true
+    """
+    validate_inputs.py '${userParamsJson}' $vcf $hpo $refDir
+
+    file_path="\${PWD}/validation_results.csv"
+
+    if [ -f "\$file_path" ]; then
+        check_validation_status.py \$file_path
+    else
+        echo "CSV file not found: \$file_path"
+        exit 1
+    fi
+    """
```

Comment:
I don’t think we need this. As we discussed, you only need to print it out when you find any missing files. There’s no need to check further once you’ve located any of them. Please remove this part then pass the value based on the result of `validate_inputs.py`

Reply:
I should've mentioned earlier this and I might confuse you, but you can simplify this like what others did:

https://github.com/nf-core/chipseq/blob/51eba00b32885c4d0bec60db3cb0a45eb61e34c5/workflows/chipseq.nf#L16-L28

Reply:
I agree, we don't have to but it's a good practice to have separation of responsibility in classes/scripts so that we don't end up with long scripts doing multiple thing in them. Hence the reason for having scripts once for validation and other to handles the printing. 

But since in the validation checks are not for a lot of parameters, I'll move the printing logic into the validate_input script. 

Saving the errors to .csv will need to stay since we'll eventually need it for our web-app users too, so that we can also display error messages on the portal, I case of job filers.

Reply:
I agree and disagree - having a good practice is important. I don't think web-app users need to know it - they were not able to handle the parameter, so the error they may need to know is after this.